scan-payment/src/main/java/com/getbouncer/scan/payment/Image.kt

```diff
@@ -81,6 +84,102 @@ fun hasOpenGl31(context: Context): Boolean {
     }
 }
 
+/**
+ * Fragments the image into multiple segments and places them in new segments. The operations on the
+ * image are run in parallel.
+ */
+suspend fun Bitmap.fragment(fromSegments: Array<Rect>,
+                            toSegments: Array<Rect>,
+                            toSize: Size): Bitmap {
+    require(fromSegments.size == toSegments.size) {
+        "Number of source segments does not match number of destination segments"
+    }
+
+    val current = this
+    val result = Bitmap.createBitmap(toSize.width, toSize.height, this.config)
+    val canvas = Canvas(result)
+
+    coroutineScope {
+        (0 until fromSegments.size).map {
+            async {
```

Comment:
async is usually used to return a job. If you're just using side-effects, we can probably change this to `launch` with a `.joinAll` at the end.

Is drawing bitmaps to a canvas thread safe? have you tried pulling that logic out of the async flow and testing the performance with it separately? If it's blocking at all, it could negatively impact the performance of the crop calculation.